agents/rust/mediator/src/didcomm_handlers/mod.rs

```diff
@@ -35,7 +34,7 @@ pub async fn handle_aries<T: BaseWallet + 'static, P: MediatorPersistence>(
     log::info!("processing message {:?}", &didcomm_msg);
     let unpacked = agent.unpack_didcomm(&didcomm_msg).await.unwrap();
     let aries_message: GeneralAriesMessage =
-        serde_json::from_str(&unpacked.message).expect("Decoding unpacked message as AriesMessage");
+        serde_json::from_str(&unpacked.message).map_err(|e| e.to_string())?;
```

Comment:
As mentioned elsewhere, consider introducing custom `MediatorError` error type, along with something like `MediatorResult<T>` as alias to `Result<T, MediatorError>`. See how we do the same in many of our other crates.

Reply:
Adding custom errors is sounds like a separate PR though :-)